src/main/java/fi/aalto/cs/apluscourses/ui/module/ModulesView.java

```diff
@@ -47,9 +64,12 @@ public JPanel getBasePanel() {
    * Update this modules view with the given view model (which may be null).
    */
   public void viewModelChanged(@Nullable CourseViewModel course) {
-    ApplicationManager.getApplication().invokeLater(
-        () -> moduleListView.setModel(course == null ? null : course.getModules()),
+    ApplicationManager.getApplication().invokeLater(() -> {
+      moduleListView.setModel(course == null ? null : course.getModules());
+      cl.show(cardPanel, (course != null) ? "TreeCard" : "LabelCard");
```

Comment:
This should probably also use `MainViewModel.getCard...` (unless the method is changed, as I mentioned in another comment)

Reply:
I agree that this should use the same method as ExercisesView. It would require passing MainViewModel to this class also but that's nice symmetry.

Reply:
Sorry nevermind, the modules view of course doesn't have a reference to the main view model

Reply:
> Sorry never mind

No! Please mind this. 😂

@nikke234 Okay it seems we have a bit of a disagreement. 😘

@stellatsv I'd like to hear what do you think.

Reply:
@OlliKiljunen Look at the comment times! :smile: I posted mine before your review, but confusingly it's shown after your comment.

In other words, I changed my mind on my own comment before seeing your comment at all.

That having been said, I would not introduce main view model to `ModulesView` (even though it is symmetric), since I think it also doesn't belong to `ExercisesView` :smile: